DigitalLearningSolutions.Web.Tests/Helpers/ProfessionalRegistrationNumberHelperTests.cs

```diff
@@ -0,0 +1,137 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.Helpers
+{
+    using System.Linq;
+    using DigitalLearningSolutions.Web.Helpers;
+    using FluentAssertions;
+    using FluentAssertions.Execution;
+    using Microsoft.AspNetCore.Mvc.ModelBinding;
+    using NUnit.Framework;
+
+    public class ProfessionalRegistrationNumberHelperTests
+    {
+        [TestCase(null)]
+        [TestCase(false)]
+        public void SetHasProfessionalRegistrationNumber_returns_null_when_has_not_been_prompted(
+            bool? hasBeenPromptedForPrn
+        )
+        {
+            // When
+            var result =
+                ProfessionalRegistrationNumberHelper.SetHasProfessionalRegistrationNumber(hasBeenPromptedForPrn, "prn");
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void SetHasProfessionalRegistrationNumber_returns_false_when_has_been_prompted_and_no_prn()
+        {
+            // When
+            var result =
+                ProfessionalRegistrationNumberHelper.SetHasProfessionalRegistrationNumber(true, null);
+
+            // Then
+            result.Should().BeFalse();
+        }
+
+        [Test]
+        public void SetHasProfessionalRegistrationNumber_returns_true_when_has_been_prompted_and_has_prn()
+        {
+            // When
+            var result =
+                ProfessionalRegistrationNumberHelper.SetHasProfessionalRegistrationNumber(true, "prn-12");
+
+            // Then
+            result.Should().BeTrue();
+        }
+
+        [TestCase(true, false)]
+        [TestCase(false, true)]
+        public void ValidateProfessionalRegistrationNumber_does_not_set_errors_when_not_delegate_or_has_no_prn(
+            bool isDelegate,
+            bool hasPrn
+        )
+        {
+            // Given
+            var state = new ModelStateDictionary();
+
+            // When
+            ProfessionalRegistrationNumberHelper.ValidateProfessionalRegistrationNumber(
+                state,
+                hasPrn,
+                null,
+                isDelegate
+            );
+
+            // Then
+            state.IsValid.Should().BeTrue();
+        }
+
+        [Test]
+        public void ValidateProfessionalRegistrationNumber_does_not_set_errors_when_valid_prn()
+        {
+            // Given
+            var state = new ModelStateDictionary();
+            const string validPrn = "abc-123";
+
+            // When
+            ProfessionalRegistrationNumberHelper.ValidateProfessionalRegistrationNumber(
+                state,
+                true,
+                validPrn
+            );
+
+            // Then
+            state.IsValid.Should().BeTrue();
+        }
+
+        [Test]
+        public void ValidateProfessionalRegistrationNumber_sets_error_when_hasPrn_is_not_set()
+        {
+            // Given
+            var state = new ModelStateDictionary();
+            var expectedError = "Select your professional registration number status.";

```

Comment:
I've realised my problem with putting expected output in `Given` is that all of the test data and mock setups go there. So when I see a string literal here I always assume it's input data. Alright if we put this in `Then`?

Reply:
Also c.f. above test where expected output of `true` has to go in `Then` because it's a fluent assertion method.

Reply:
Done.